magento/magento2
# issue with the price display in the Recently Viewed widget

### Title
Out-of-stock simple product prices are not shown in the “Recently Viewed Product” widget

### Description (*)
When we visit the out-of-stock simple product in the frontend that product price is not shown in the “Recently viewed Product” widget however we can see the same out-of-stock product price in the “Related Product” Widget And “Product list page” so as per Magento behavior the price of the out of stock simple product should be shown in “Recently viewed Product” Widget also.in screenshot1 you can see that we can not see the price in the ‘recently viewed product’ widget but  we can see the same ‘out-of-stock’ product price in the ‘related product’ widget 
Screenshot 1:
![image](https://github.com/magento/magento2/assets/124759248/8a47e1ff-9ae1-4117-87d3-adfac76cdc41)

### Manual testing scenarios (*)
1: Access the Magento admin panel and navigate to 'content' -> 'Widget' and   add a widget
2: in “Storefront Properties” and “Layout Updates” select  the option as shown in screenshot2
Screenshot 2:
![image](https://github.com/magento/magento2/assets/124759248/7d3c5e49-ff45-4801-9cfb-1273d7433269)

3:in the widget options setting select the option as shown in below screenshot and save the widget
Screenshot 3:
![image](https://github.com/magento/magento2/assets/124759248/13a07ea3-f9cc-4c15-a32e-fd6ac90d6459)

4: create some simple product and set them as ‘Out of stock’ or change the status of existing products to ‘Out of stock’

5: After navigate to 'stores' -> 'configuration'->catalog->inventory in stock option tab set the ‘display out of stock product’ to yes as you can see in screenshot 4
![image](https://github.com/magento/magento2/assets/124759248/bff1841e-29c3-4f18-98dc-139e65f3f29f)

6:now you visit these ‘out-of-stock’ simple product and  see the recently viewed product widget you can see that the price of out of stock simple product is not shown in the “Recently Viewed Product”
Screenshot 5:
![image](https://github.com/magento/magento2/assets/124759248/31a0697e-990a-4842-aa04-fff787a2820a)

### Expected result 
The price of out-of-stock simple product should be shown in the Recently Viewed Product Widget 

###  Actual result 
we can not see the price of the out-of-stock simple products in the Recently Viewed Product Widget

### Additional Information 

Recently viewed product widget load Price from these two HTML files: ‘regular_price.html’ and ‘special_price.html’, located in the following path:
1:magento/module-catalog/view/base/web/template/product/price/regular_price.html
2:magento/module-catalog/view/base/web/template/product/price/special_price.html

both files contain a condition on line 7: <if args="isSalable($row())"> this condition determines whether the product is salable(in-stock). If the product is not salable(out-of-stock), the price of that product will not be displayed in the “Recently Viewed Product” Widget. These templates only load prices for the “Recently Viewed Product” widget; therefore, you can view the price of the out-of-stock simple products in the ‘Related Product’ and “Product List” pages

I have included a screenshot(screenshot 6) of the regular_price.html file so you can see the condition in line 7 of that file 
Screenshot 6:
![image](https://github.com/magento/magento2/assets/124759248/c32022aa-595f-44c5-8ff8-1faa2f916212)

### Solution 

So to see the out-of-stock simple product price in the recently viewed product widget we have to modify the regular_price.html file and  special_price.html and remove the <if  "isSalable($row())"> condition so after applying this we can see the price of the out-of-stock simple product price in “Recently Viewed Product” widget. i have added the screenshot(screenshot 7) of the regular_price.html file for your reference so you can see the file after the changes

Screenshot 7:
![image](https://github.com/magento/magento2/assets/124759248/7b312981-52f8-45c0-9cb7-9797058828c6)


### Resolved issues:
1. [x] resolves magento/magento2#38167:  issue with the price display in the Recently Viewed widget


### Related Pull Requests
https://github.com/magento-gl/magento2-page-builder/pull/75
